src/toHaveStyleRule.js

```diff
@@ -20,10 +20,18 @@ const getClassNames = received => {
 const hasClassNames = (classNames, selectors) =>
   classNames.some(className => selectors.includes(`.${className}`))
 
-const getRules = (ast, classNames) =>
-  ast.stylesheet.rules.filter(
+const getRules = (ast, classNames, options) => {
+  let rules = ast.stylesheet.rules
+  if (options.media) {
+    rules = rules
```

Comment:
Two questions for you:

1. What do you think about moving this into a separate function?

Something like:
```js
const rules = options.media ? getMediaRules(ast, options.media) : ast.stylesheet.rules
```

2. What if we make this function even more generic?

```js
const getAtRules = (ast, options) => Object.keys(options).map(option =>
  ast.stylesheet.rules.filter(rule => rule.type === option && rule[option] === options[option])
    .map(rule => rule.rules)
    .reduce((acc, rules) => acc.concat(rules), [])
)
```

It would support `@supports` for free :)